### PR TITLE
Add controller for case transformation

### PR DESCRIPTION
**Pull request by rkondratowicz**
Still to do (in next PRs):
- set up json datetime serialization and deserialization
- wrap scannable items in an object (for CCD)

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/controllers/TransformationControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    @Test
+    public void should_return_proper_status_codes_for_auth_exceptions() throws Exception {
+        List<Pair<RuntimeException, HttpStatus>> exceptionsAndStatuses =
+            asList(
+                Pair.of(new UnauthenticatedException(null), UNAUTHORIZED),
+                Pair.of(new InvalidTokenException(null, null), UNAUTHORIZED),
+                Pair.of(new ForbiddenException(null), FORBIDDEN)
+            );
+
+        for (Pair<RuntimeException, HttpStatus> pair : exceptionsAndStatuses) {
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/controllers/TransformationControllerTest.java]: mockMvc throws checked exceptions, can call `.foreach(x -> ...)` on list above 😢

File: src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/controllers/TransformationControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
+package uk.gov.hmcts.reform.bulkscanccdeventhandler.transformation.controllers;
+
+import com.microsoft.applicationinsights.web.dependencies.apachecommons.lang3.tuple.Pair;
```
Comment (doncem) [inline src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/controllers/TransformationControllerTest.java]: Is this correct?

File: src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/controllers/TransformationControllerTest.java (hunk 1 above)
Comment (doncem) [reply src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/controllers/TransformationControllerTest.java]: How about extracting loop source into parameterised test source?

```java
private static Stream<Arguments> exceptionsAndStatuses() {
    return Stream.of(
      Arguments.of(new UnauthenticatedException(null), UNAUTHORIZED),
      Arguments.of(new InvalidTokenException(null, null), UNAUTHORIZED),
      Arguments.of(new ForbiddenException(null), FORBIDDEN)
    );
}

@ParameterizedTest
@MethodSource("exceptionsAndStatuses")
public void should_return_proper_status_codes_for_auth_exceptions(
    RuntimeException exception,
    HttpStatus expectedStatus
) throws Exception {
    reset(authService);
    given(authService.authenticate(any())).willThrow(exception);

    sendRequest("{}").andExpect(status().is(expectedStatus.value()));
}
```

can easily extend the source

Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/controllers/TransformationControllerTest.java]: done

File: src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/controllers/TransformationControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        Mockito.reset(authService);
+    }
+
+    @ParameterizedTest
+    @MethodSource("exceptionsAndStatuses")
+    public void should_return_proper_status_codes_for_auth_exceptions(RuntimeException exc, HttpStatus status) throws Exception {
+        given(authService.authenticate(any())).willThrow(exc);
+
+        sendRequest("{}")
+            .andExpect(status().is(status.value()));
```
Comment (doncem) [inline src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/controllers/TransformationControllerTest.java]: i thought compiler would've complained about `status` usage here :D

File: src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/controllers/TransformationControllerTest.java (hunk 2 above)
Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/controllers/TransformationControllerTest.java]: oops

Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/transformation/controllers/TransformationControllerTest.java]: gone now

**Pull request merged**